### PR TITLE
Remove configuration concept

### PR DESCRIPTION
**Pull request by crisu83**
After I changed the AST nodes to use constructor params when I was optimizing the parser I realized that the same practice could be applied to project-wide. That is why I refactored both the types and schema to use constructor params and removed the config package completely.

### CODE REVIEW

File: src/Type/Definition/FieldsTrait.php [hunk 1]
```diff
@@ -77,14 +80,14 @@ protected function buildFieldMap($fieldsOrThunk): array
...
-                \sprintf('%s.%s field config must be an array', $this->getName(), $fieldName)
+                \sprintf('%s.%s field config must be an array', $this->name, $fieldName)
             );
 
             invariant(
                 !isset($fieldConfig['isDeprecated']),
                 \sprintf(
                     '%s.%s should provide "deprecationReason" instead of "isDeprecated".',
-                    $this->getName(),
+                    $this->name,
```
Comment (hungneox) [inline src/Type/Definition/FieldsTrait.php]: @crisu83 Why did you change `$this->getName()` to `$this->name`? :D Plus there is no `$name` attribute in this trait.

File: src/Type/Definition/FieldsTrait.php [hunk 2]
```diff
@@ -94,15 +97,22 @@ protected function buildFieldMap($fieldsOrThunk): array
...
                         toString($fieldConfig['resolve'])
                     )
                 );
             }
 
-            $fieldConfig['name'] = $fieldName;
-            $fieldMap[$fieldName] = new Field($fieldConfig);
+            $fieldMap[$fieldName] = new Field(
+                $fieldName,
+                $fieldConfig['description'] ?? null,
```
Comment (hungneox) [inline src/Type/Definition/FieldsTrait.php]: I think maybe we wnat to get rid of `ConfigAwareTrait` but this kind of workaround smells bad to me :-?

File: src/Type/Definition/Field.php [hunk 3]
```diff
@@ -2,19 +2,51 @@
...
+
+    /**
+     * Field constructor.
+     *
+     * @param string                                                       $name
+     * @param null|string                                                  $description
+     * @param TypeInterface|OutputTypeInterface|WrappingTypeInterface|null $type
+     * @param Argument[]|array[]                                           $arguments
+     * @param callable|null                                                $resolve
+     * @param null|string                                                  $deprecationReason
```
Comment (hungneox) [inline src/Type/Definition/Field.php]: I think if you let PHPStorm generate phpdoc it should be `string|null`.

File: src/Type/Definition/FieldsTrait.php (hunk 1 above)
Comment (crisu83) [reply src/Type/Definition/FieldsTrait.php]: I fixed this.

File: src/Type/Definition/EnumType.php [hunk 4]
```diff
@@ -21,7 +21,7 @@
  *
  * Example:
  *
- *     $RGBType = new GraphQLEnumType([
+ *     $RGBType = new newEnumType([
```
Comment (fubhy) [inline src/Type/Definition/EnumType.php]: That seems wrong.

File: src/Type/Definition/EnumType.php [hunk 5]
```diff
@@ -209,7 +199,7 @@ protected function buildValues(array $valueMap): array
                 isAssocArray($valueConfig),
                 \sprintf(
                     '%s.%s must refer to an object with a "value" key representing an internal value but got: %s',
-                    $this->getName(),
```
[the remark this answers is not in the document]
Comment (crisu83) [reply src/Type/Definition/EnumType.php]: Only for traits.

File: src/Type/Definition/EnumType.php [hunk 6]
```diff
@@ -219,7 +209,7 @@ protected function buildValues(array $valueMap): array
                 !isset($valueConfig['isDeprecated']),
                 \sprintf(
                     '%s.%s should provided "deprecationReason" instead of "isDeprecated".',
-                    $this->getName(),
+                    $this->name,
```
[the remark this answers is not in the document]
Comment (crisu83) [reply src/Type/Definition/EnumType.php]: I'd prefer to use fields directly where possible, in traits we cannot do that for obvious reasons.
